main.go

```diff
@@ -111,6 +116,16 @@ func main() {
 	}
 
 	if socket != "" {
+		go func() {
+			for {
+				if ! isParentAlive() {
+					os.Exit(0)
+				}
+
+				time.Sleep(1 * time.Minute)
```

Comment:
I think we can poll much more frequently, what about 1 seconds? There is basically negligible performance impact from this logic. Exiting the plugin server too slowly could cause excessive memory usage under certain cases (for example, when user restarts Kong many times over a short time period).